### PR TITLE
wolfstoneextract: init at 1.2

### LINKED ISSUE #213363: Package request: WolfstoneExtract
**Project description**
Buried inside the game files for [Wolfenstein II: The New Colossus](https://bethesda.net/en/games/product/WO5CSTPCBG01) and [Wolfenstein: Youngblood](https://bethesda.net/en/games/product/WOYCD1PCBG01BASE) is game data for two new [Wolfenstein 3D](https://bethesda.net/en/games/product/WO1CSTPCBG01)-engine games, [Wolfstone 3D and Elite Hans: Die Neue Ordnung](https://wolfenstein.fandom.com/wiki/Wolfstone_3D). WolfstoneExtract allows you to convert the embedded Wolfstone 3D and Elite Hans: Die Neue Ordnung game data into a format that’s usable by [ECWolf](https://maniacsvault.net/ecwolf/).

**Metadata**

* homepage URL: <https://bitbucket.org/ecwolf/wolfstoneextract>
* source URL: <https://bitbucket.org/ecwolf/wolfstoneextract>
* license: GPLv3 and various permissive free licenses
* platforms: Linux

### PR DESCRIPTION
**Pull request by keenanweaver**
###### Description of changes

Adds wolfstoneextract 1.2 to nixpkgs. Resolves https://github.com/NixOS/nixpkgs/issues/213363. Tested utiility on the GOG version and it extracted fine.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
Would you please rebase it to take in the maintainer's information?
I think the ofborg error will disappear now that #239974 has been merged.

Then please use `nixpkgs-fmt` as a formatter for now.

File: pkgs/games/wolfstoneextract/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  meta = with lib; {
+    description = "Utility to extract Wolfstone data from Wolfenstein II";
+    homepage = "https://bitbucket.org/ecwolf/wolfstoneextract/src/master/";
+    license = licenses.gpl3Only;
```
Comment (natsukium) [inline pkgs/games/wolfstoneextract/default.nix:24]:
```suggestion
    license = with licenses; [ gpl3Only bsd3 ];
```
It seems to include bsd3.
https://bitbucket.org/ecwolf/wolfstoneextract/src/e1d37f2ad5b5207431fa6328cc0f6602960ebf66/src/doomerrors.h#lines-5:31

File: pkgs/games/wolfstoneextract/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ lib
+, stdenv
+, fetchFromBitbucket
+, cmake
+}:
+stdenv.mkDerivation rec {
```
Comment (natsukium) [inline pkgs/games/wolfstoneextract/default.nix:6]:
```suggestion
}:

stdenv.mkDerivation rec {
```
please add a blank line here.

File: pkgs/games/wolfstoneextract/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{ lib
+, stdenv
+, fetchFromBitbucket
+, cmake
+}:
+
+stdenv.mkDerivation rec {
```
Comment (drupol) [inline pkgs/games/wolfstoneextract/default.nix:7]: I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/games/wolfstoneextract/default.nix (hunk 3 above)
Comment (keenanweaver) [reply pkgs/games/wolfstoneextract/default.nix]: Thanks for the tip! I'll do some reading. In the meantime, I've added your suggestion & successfully built it.

**Review by drupol — APPROVED** [review]

**Review by natsukium — APPROVED** [review]
It builds fine on my x86_64-linux and will probably work. (I don't have Wolfenstein II installed.)

**Review by drupol — APPROVED** [review]

**Pull request merged**